dags/openshift_nightlies/scripts/install/rosa.sh

```diff
@@ -24,23 +24,81 @@ _get_cluster_status(){
     echo $(rosa list clusters -o json | jq -r '.[] | select(.name == '\"$1\"') | .status.state')
 }
 
+_wait_for_nodes_ready(){
+    _download_kubeconfig $(_get_cluster_id $1) ./kubeconfig
+    export KUBECONFIG=./kubeconfig
+    ALL_READY_ITERATIONS=0
+    ITERATIONS=0
+    # Node count is number of workers + 3 masters + 3 infra
+    NODES_COUNT=$(($2+6))
+    # 180 seconds per node, waiting 5 times 60 seconds (5*60 = 5 minutes) with all nodes ready to finalize 
+    while [ ${ITERATIONS} -le ${NODES_COUNT} ] ; do
+        NODES_READY_COUNT=$(oc get nodes | grep " Ready " | wc -l)
+	if [ ${NODES_READY_COUNT} -ne ${NODES_COUNT} ] ; then
+	    echo "WARNING: ${ITERATIONS}/${NODES_COUNT} iterations. ${NODES_READY_COUNT}/${NODES_COUNT} nodes ready. Waiting 180 seconds for next check"
+            ALL_READY_ITERATIONS=0
+            ITERATIONS=$((${ITERATIONS}+1))
+            sleep 180
+        else
+	    if [ ${ALL_READY_ITERATIONS} -eq 5 ] ; then
+                echo "INFO: ${ALL_READY_ITERATIONS}/5. All nodes ready, continuing process"
+                return 0
+            else
+                echo "INFO: ${ALL_READY_ITERATIONS}/5. All nodes ready. Waiting 60 seconds for next check"
+                ALL_READY_ITERATIONS=$((${ALL_READY_ITERATIONS}+1))
+                sleep 60
+	    fi
+	fi
+    done
+    echo "ERROR: Not all nodes (${NODES_READY_COUNT}/${NODES_COUNT}) are ready after about $((${NODES_COUNT}*3)) minutes, dumping oc get nodes..."
+    oc get nodes
+    exit 1
+}
+
 _wait_for_cluster_ready(){
-    echo "INFO: Waiting about 3 hour for cluster on ready status"
+    START_TIMER=$(date +%s)
+    echo "INFO: Installation starts at $(date -d @${START_TIMER})"
+    echo "INFO: Waiting about 180 iterations, counting only when cluster enters on installing status"
+    ITERATIONS=0
+    PREVIOUS_STATUS=""
     # 180 iterations, sleeping 60 seconds, 3 hours of wait
-    for i in `seq 1 180` ; do
+    # Only increasing iterations on installing status
+    while [ ${ITERATIONS} -le 180 ] ; do
```

Comment:
we could cut this time in half since a OCP install (with day 2 ops excluded) should not exceed 90 minutes for sure

Reply:
moved to 90